fluent-syntax/src/parser.ts

```diff
@@ -669,7 +667,9 @@ export class FluentParser {
     return selector;
   }
 
-  getInlineExpression(ps: FluentParserStream): AST.Expression | AST.Placeable {
+  getInlineExpression(
+    ps: FluentParserStream
+  ): AST.InlineExpression | AST.Placeable {
```

Comment:
`suggestion (if-minor)`: this could be a bit cleaner if we separated `getPlaceable` from `getInlineExpression`.

Reply:
I gave this a look, and my intuition isn't strong here. And I don't think our tests are exhaustive. Like, I was rather surprised that 

```
shared-photos = {NUMBER({1 ->
  *[1] 2
})}
```

works. Which made me wonder if it's not about expressions when it comes down to block vs inline, but placeable.

I've paged most of that context out of my brain, so it's not minor to me.

Playground also seems to behave differently to my local repo, and I'll call shotgun on that one. It still uses the `0.14` compat build.

Reply:
ouch. That's not spec compliant, right?

Spec doesn't allow for it - https://github.com/projectfluent/fluent/blob/master/spec/fluent.ebnf#L74
Rust doesn't allow for it - https://github.com/projectfluent/fluent-rs/blob/master/fluent-syntax/src/parser/expression.rs#L176

maybe it's time to consider Rust->JS export for the parser?

Reply:
I think the spec allows it, https://github.com/projectfluent/fluent/blob/master/spec/fluent.ebnf#L59 has the `inline_placeable` as inline expression. And I tested the fluent-rs parser, too, and it was fine.

```
jazz = {{{1->
  *[1] 2
  }}->
  *[1] 3
}
```

OTH is illegal. Works on playground as in not only in the resolver, but also shows an AST.

Reply:
ah, right. The `placeholder` loophole in inline :) retracting my previous statement.